DB Design

hirojiang opened the pull request:
#WHAT
・Users, Groups, Members, Messagesテーブル設計
・それぞれのカラム/アソシエーション情報を記載

#WHY
必要なテーブル、カラムの作成の為。

Commit:
・DB Design (users, groups, members, messages) table added to README.md file

MrUrakou commented on README.md:
```diff
@@ -22,3 +22,54 @@ Things you may want to cover:
+
+|Column|Type|Options|
+|------|----|-------|
+|body|text|null: false|
```
bodyかimageがあれば投稿できる仕様なので、null: false制約は不要です。

MrUrakou commented on README.md:
```diff
@@ -22,3 +22,54 @@ Things you may want to cover:
+|Column|Type|Options|
+|------|----|-------|
+|body|text|null: false|
+|image|string|default: " "|
```
default: " "って必要なのですか？

MrUrakou reviewed (commented):
全体的にアソシエーションを見直しましょう。
また、user_idとgroup_idはreference型にしましょう。
参考url
https://qiita.com/ryouzi/items/2682e7e8a86fd2b1ae47

Commit:
Revised DB Design based on review

・changed the table name from “members” to “group_users” with all
related place
・use the “reference” type to define group_users table
・revise association about messages
・delete [null: false] at body column
・delete [default: “ ”]

hirojiang commented:
#What
【アソシエーション見直し】
・message は users, groups 両テーブルに属する。両テーブル情報を紐付けた group_users テーブルが既に存在する為 messages テーブルへ → [belongs_to :group_user] を追加
・users, groups テーブルは複数の message を持つため 両テーブルへ → [has_many :messages, through: :group_users] を追加

【指摘箇所】
・group_users テーブルを reference 型へ変更
・body カラム [null: false] を削除
・image カラム [default: " "] を削除

【その他】
・可読性向上の為、中間テーブル名変更(同名箇所も全変更) [members] → [group_users]

#Why
・レビュー後のDB設計修正のため

miyagit commented on README.md:
```diff
@@ -46,30 +47,29 @@ Things you may want to cover:
+- has_many :messages through: :group_users
+- has_many :users, through: :group_users
+- has_many :group_users
+- accepts_nested_attributes_for:group_users
```
コレ使わないですね。

Commit:
Revised DB Design based on review_v3

・deleted [accepts_nested_attributes_for:group_users]
・Separated association [belongs_to :group_user] at messages table to
[belongs_to :user] & [belongs_to :group]
・deleted [through: :group_users] from both association at users table &
groups table

hirojiang commented:
#What
⑴ ReadmeファイルへのDB設計記載の為、groupsテーブル [accepts_nested_attributes_for:group_users]を削除
⑵ messageテーブルで [belongs_to :group_user
] → [belongs_to :user
] [belongs_to :group] へ分解
⑶ users, groups両テーブルで [through: :group_users] 削除（messageとuser, groupは多対多でない為）

#Why
・レビュー後のDB設計修正のため

yamamotokeishiro commented:
修正出来ております！
